pkg/storer/compact_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storer_test
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/log"
+	"github.com/ethersphere/bee/pkg/postage"
+	postagetesting "github.com/ethersphere/bee/pkg/postage/testing"
+	pullerMock "github.com/ethersphere/bee/pkg/puller/mock"
+	chunk "github.com/ethersphere/bee/pkg/storage/testing"
+	storer "github.com/ethersphere/bee/pkg/storer"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// TestCompact creates two batches and puts chunks belonging to both batches.
+// The first batch is then expired, causing free slots to accumulate in sharky.
+// Next, sharky is compacted, after which, it is tested that valid chunks can still be retrieved.
+func TestCompact(t *testing.T) {
```

Comment:
this is quite a strange test for compaction. I mean it is totally unnecessary to have such a complex test. Why should compaction know about batches or eviction etc.